### PR TITLE
add Pinky(https://github.com/tamanishi/Pinky3) keyboard

### PR DESCRIPTION
**Pull request by tamanishi**
This PR adds my new "Pinky" keyboard firmware.

## Description
Add my new "Pinky" keyboard firmware.  
- [Pinky3(3rows and thumb keys)](https://github.com/tamanishi/Pinky3)
- [Pinky4(4rows and thumb keys)](https://github.com/tamanishi/Pinky4)

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR

* none

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
If you have two different revisions of the PCB (3 and 4 pinky), you should really use these as the "revision", instead of using a configuration in the keymap (and rules.mk).

Put most everything in the main folder, but change the pin defines in the subfolders, and then have a seperate keymap folder in each of these revisions.  

It would be a lot cleaner here.

**Comment by tamanishi** [comment]
Thank you for reviewing.
I separated revisions 3 and 4.

File: keyboards/pinky/3/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    // set_timelog();
+  }
+
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_QWERTY);
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/pinky/3/keymaps/default/keymap.c]: Also, if you only have the single layer, you should probably remove this macro, entirely.

File: keyboards/pinky/3/3.c [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#include "pinky.h"
```
Comment (drashna) [inline keyboards/pinky/3/3.c]: Actually, please remove everything but this line. 

The rest of this is handled by core code and should not need to be called here. 

https://docs.qmk.fm/#/feature_audio

File: keyboards/pinky/4/4.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include "pinky.h"
```
Comment (drashna) [inline keyboards/pinky/4/4.c]: Actually, please remove everything but this line. 

The rest of this is handled by core code and should not need to be called here. 

https://docs.qmk.fm/#/feature_audio

File: keyboards/pinky/3/config.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+/* key matrix size */
+// Rows are doubled-up
+#define MATRIX_ROWS 8
+#define MATRIX_COLS 7
+#define MATRIX_ROW_PINS { D4, C6, D7, E6 }
+
+// wiring of each half
+#define MATRIX_COL_PINS { F4, F5, F6, F7, B1, B3, B2 }
+// #define MATRIX_COL_PINS { B2, B3, B1, F7, F6, F5, F4 } //uncomment this line and comment line above if you need to reverse left-to-right key order
+
```
Comment (drashna) [inline keyboards/pinky/3/config.h]: You need `#define SOFT_SERIAL_PIN D2` here, somewhere.

File: keyboards/pinky/4/config.h [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+/* key matrix size */
+// Rows are doubled-up
+#define MATRIX_ROWS 10
+#define MATRIX_COLS 7
+#define MATRIX_ROW_PINS { D4, C6, D7, E6, B4 }
+
+// wiring of each half
+#define MATRIX_COL_PINS { F4, F5, F6, F7, B1, B3, B2 }
+// #define MATRIX_COL_PINS { B2, B3, B1, F7, F6, F5, F4 } //uncomment this line and comment line above if you need to reverse left-to-right key order
+
```
Comment (drashna) [inline keyboards/pinky/4/config.h]: You need `#define SOFT_SERIAL_PIN D2` here, somewhere.

File: keyboards/pinky/config.h [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+#define USE_I2C
+#define USE_SERIAL
+
+#ifdef USE_Link_Time_Optimization
+  // LTO has issues with macros (action_get_macro) and "functions" (fn_actions),
+  //  so just disable them
+  #define NO_ACTION_MACRO
+  #define NO_ACTION_FUNCTION
+
+  #define DISABLE_LEADER
```
Comment (drashna) [inline keyboards/pinky/config.h]: This isn't used anymore, and can be removed

**Comment by tamanishi** [comment]
Happy New Year!
Thank you for reviewing.
I hope that I improved the items was pointed out above.

File: keyboards/pinky/pro_micro.h [hunk 7]
```diff
@@ -0,0 +1,358 @@
+/*
```
Comment (drashna) [inline keyboards/pinky/pro_micro.h]: I don't think that this file is needed, and should be removed.

File: keyboards/pinky/rules.mk [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+CONSOLE_ENABLE = no         # Console for debug(+400)
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
+SUBPROJECT_rev1 = no
```
Comment (drashna) [inline keyboards/pinky/rules.mk]: This is a legacy setting and should be removed.

```suggestion
```

File: keyboards/pinky/rules.mk [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+USE_I2C = yes
```
Comment (drashna) [inline keyboards/pinky/rules.mk]: This isn't actually used, anywhere and should be removed. 

```suggestion
```

File: keyboards/pinky/rules.mk [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+CUSTOM_MATRIX = yes
```
Comment (drashna) [inline keyboards/pinky/rules.mk]:
```suggestion
SPLIT_KEYBOARD = yes
```

File: keyboards/pinky/pro_micro.h (hunk 7 above)
Comment (tamanishi) [reply keyboards/pinky/pro_micro.h]: I removed that file.

File: keyboards/pinky/rules.mk (hunk 8 above)
Comment (tamanishi) [reply keyboards/pinky/rules.mk]: I removed that.

File: keyboards/pinky/rules.mk (hunk 9 above)
Comment (tamanishi) [reply keyboards/pinky/rules.mk]: I removed that.

File: keyboards/pinky/rules.mk (hunk 10 above)
Comment (tamanishi) [reply keyboards/pinky/rules.mk]: I modified as above.

File: keyboards/pinky/3/3.h [hunk 11]
```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "../pinky.h"
```
Comment (drashna) [inline keyboards/pinky/3/3.h]: You don't actually need the ".." here. 

When compiling, both the subfolder and main folder (eg, pinky, and 3 in this case) are added to the "common path".  So, you can call "pinky.h" directly, without having to use it's relative path. 

It's fin to do this, but "for the future"... it's something to keep in mind.

**Comment by drashna** [comment]
Awesome, thanks!

File: keyboards/pinky/3/3.h (hunk 11 above)
Comment (tamanishi) [reply keyboards/pinky/3/3.h]: I understood. I'll keep that in mind.

**Pull request merged**